src/overlays/actors/ovl_En_Ma2/z_en_ma2.c

```diff
@@ -19,39 +30,382 @@ const ActorInit En_Ma2_InitVars = {
+        1.0f,
+        0x00,
+        -10.0f,
+    },
+    {
+        0x060093BC,
+        1.0f,
+        0x00,
+        0.0f,
+    },
+    {
+        0x06009EE0,
+        1.0f,
+        0x00,
+        0.0f,
+    },
+    {
+        0x06009EE0,
+        1.0f,
+        0x00,
+        -10.0f,
+    },
+};
+
+Vec3f D_80AA28A8 = { 900.0f, 0.0f, 0.0f };
+
+u32 D_80AA28B4[] = {
+    0x06002970,
+    0x06003570,
+    0x06003770,
+};
+u32 D_80AA28C0[] = {
+    0x06002570,
+    0x06002C70,
+    0x06003070,
+};
+
+extern u32 D_06005420;
+extern SkeletonHeader D_06008D90;
+extern AnimationHeader D_060093BC;
+extern AnimationHeader D_06009EE0;
+
+u16 func_80AA19A0(GlobalContext* globalCtx, Actor* this) {
+    u16 faceReaction = Text_GetFaceReaction(globalCtx, 23);
+    if (faceReaction != 0) {
+        return faceReaction;
+    }
+    if (gSaveContext.eventChkInf[1] & 0x100) {
+        return 0x2056;
+    }
+    if (gSaveContext.nightFlag == 1) {
+        if (gSaveContext.infTable[8] & 0x1000) {
+            return 0x2052;
+        } else if (gSaveContext.infTable[8] & 0x4000) {
+            return 0x2051;
+        } else {
+            return 0x2050;
+        }
+    }
+    return 0x204C;
+}
+
+s16 func_80AA1A38(GlobalContext* globalCtx, Actor* this) {
+    s16 ret = 1;
+
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 2:
+            switch (this->textId) {
+                case 0x2051:
+                    gSaveContext.infTable[8] |= 0x1000;
+                    ret = 2;
+                    break;
+                case 0x2053:
+                    gSaveContext.infTable[8] |= 0x2000;
+                    ret = 0;
+                    break;
+                default:
+                    ret = 0;
+                    break;
+            }
+            break;
+        case 0:
+        case 1:
+        case 3:
+        case 4:
+        case 5:
+        case 6:
+        case 7:
+        case 9:
+            break;
+    }
+    return ret;
+}
+
+void func_80AA1AE4(EnMa2* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 phi_a3;
+
+    if ((this->unk_1E0.unk_00 == 0) && (this->skelAnime.animCurrentSeg == &D_06009EE0)) {
+        phi_a3 = 1;
+    } else {
+        phi_a3 = 0;
+    }
+
+    this->unk_1E0.unk_18 = player->actor.posRot.pos;
+    this->unk_1E0.unk_14 = 0.0f;
+
+    func_80034A14(&this->actor, &this->unk_1E0, 0, phi_a3);
+}
+
+u16 func_80AA1B58(EnMa2* this, GlobalContext* globalCtx) {
+    if (gSaveContext.linkAge == 1) {
+        return 0;
+    }
+    if ((!(gSaveContext.eventChkInf[1] & 0x100)) && (globalCtx->sceneNum == 0x36) && (gSaveContext.nightFlag == 0) &&
+        (this->actor.shape.rot.z == 5)) {
+        return 1;
+    }
+    if ((!(gSaveContext.eventChkInf[1] & 0x100)) && (globalCtx->sceneNum == 0x63) && (gSaveContext.nightFlag == 1) &&
+        (this->actor.shape.rot.z == 6)) {
+        return 2;
+    }
+    if ((!(gSaveContext.eventChkInf[1] & 0x100)) || (globalCtx->sceneNum != 0x63)) {
```

Comment:
Scene enum values should be used here.

Reply:
like SCENE_ZERO, SCENE_ONE, ???

Reply:
`globalCtx->sceneNum == SCENE_...`

Reply:
ohhhh psh, those scenes. oops. I thought you meant to create a new scene enum for this actor and use it as the return values. I'll do this in a bit